src/librustc_mir/transform/const_prop.rs

```diff
@@ -151,11 +152,19 @@ impl<'tcx> MirPass<'tcx> for ConstProp {
 struct ConstPropMachine<'mir, 'tcx> {
     /// The virtual call stack.
     stack: Vec<Frame<'mir, 'tcx, (), ()>>,
+    /// `OnlyInsideOwnBlock` locals that were written in the current block get erased at the end.
+    written_only_inside_own_block_locals: FxHashSet<Local>,
```

Comment:
I was going to suggest making this a bitset but I guess that won't work with the `std::mem::take` since it doesn't `impl Default`.

Reply:
`BitSet` is also always big enough to store all locals while `HashSet` resizes as required. I think it's going to be rare that `HashSet` will grow to contain a significant number of the locals. With perf clean, let's ignore this point :laughing: